src/coreclr/vm/interpexec.cpp

```diff
@@ -101,19 +128,20 @@ void InvokeDelegateInvokeMethod(MethodDesc *pMDDelegateInvoke, int8_t *pArgs, in
     pHeader->Invoke(pHeader->Routines, pArgs, pRet, pHeader->TotalStackSize);
 }
 
-void InvokeCalliStub(PCODE ftn, CallStubHeader *stubHeaderTemplate, int8_t *pArgs, int8_t *pRet)
+void InvokeCalliStub(PCODE ftn, void *cookie, int8_t *pArgs, int8_t *pRet)
```

Comment:
Why are we stripping the type information from this argument and then casting it back to CallStubHeader?

Reply:
Beacuse this API is used in an agnostic manner. For the `CallStubGenerator` scenario it is a `CallStubHeader`, when `PortableEntryPoint`s are used it is currently a staticly defined thunk. See the other implementation in this PR's `src/coreclr/vm/wasm/helpers.cpp`.